test/e2e/set/scyllacluster/scyllacluster_pv.go

```diff
@@ -5,35 +5,272 @@ package scyllacluster
+					o.Expect(err).NotTo(o.HaveOccurred())
+				case watch.Modified:
+					pvc := e.Object.(*corev1.PersistentVolumeClaim)
+					if pvc.Spec.StorageClassName == nil {
+						return false, nil
+					}
+					if pvc.DeletionTimestamp != nil {
+						return false, nil
+					}
+					if *pvc.Spec.StorageClassName != testStorageClassName && len(pvc.Spec.VolumeName) != 0 {
+						realPvc, err := f.KubeClient().CoreV1().PersistentVolumeClaims(f.Namespace()).Get(ctx, strings.TrimPrefix(pvc.Name, "clone-"), metav1.GetOptions{})
+						o.Expect(err).NotTo(o.HaveOccurred())
+
+						if len(realPvc.Spec.VolumeName) != 0 {
+							return false, nil
+						}
+
+						pv, err := f.KubeAdminClient().CoreV1().PersistentVolumes().Get(ctx, pvc.Spec.VolumeName, metav1.GetOptions{})
+						o.Expect(err).NotTo(o.HaveOccurred())
+
+						pvClone := &corev1.PersistentVolume{
+							ObjectMeta: metav1.ObjectMeta{
+								Name: fmt.Sprintf("clone-%s", pv.Name),
+								Labels: map[string]string{
+									pvCloneLabelKey: f.Namespace(),
+								},
+							},
+							Spec: corev1.PersistentVolumeSpec{
+								Capacity:                      pv.Spec.Capacity,
+								PersistentVolumeSource:        pv.Spec.PersistentVolumeSource,
+								AccessModes:                   pv.Spec.AccessModes,
+								PersistentVolumeReclaimPolicy: pv.Spec.PersistentVolumeReclaimPolicy,
+								MountOptions:                  pv.Spec.MountOptions,
+								VolumeMode:                    pv.Spec.VolumeMode,
+								StorageClassName:              testStorageClassName,
+								NodeAffinity:                  nil,
+							},
+							Status: corev1.PersistentVolumeStatus{
+								Phase: pv.Status.Phase,
+							},
+						}
+
+						if pvClone.Spec.PersistentVolumeSource.Local != nil {
+							pvClone.Spec.PersistentVolumeSource.HostPath = &corev1.HostPathVolumeSource{
+								Path: pvClone.Spec.PersistentVolumeSource.Local.Path,
+							}
+							pvClone.Spec.PersistentVolumeSource.Local = nil
+						}
+						framework.Infof("Creating clone PV for %q", pv.Name)
+						_, err = f.KubeAdminClient().CoreV1().PersistentVolumes().Create(ctx, pvClone, metav1.CreateOptions{})
+						o.Expect(err).NotTo(o.HaveOccurred())
+
+						realPvcCopy := realPvc.DeepCopy()
+						realPvcCopy.Spec.VolumeName = pvClone.Name
+
+						patch, err := helpers.CreateTwoWayMergePatch(realPvc, realPvcCopy)
+						o.Expect(err).NotTo(o.HaveOccurred())
+
+						framework.Infof("Binding ScyllaCluster PVC %q with clone PV %q", realPvc.Name, pvClone.Name)
+						_, err = f.KubeClient().CoreV1().PersistentVolumeClaims(realPvc.Namespace).Patch(ctx, realPvc.Name, types.StrategicMergePatchType, patch, metav1.PatchOptions{})
+						o.Expect(err).NotTo(o.HaveOccurred())
+					}
+				case watch.Deleted:
+					// If original PVC is deleted, delete the clone too.
+					pvc := e.Object.(*corev1.PersistentVolumeClaim)
+					if !strings.HasPrefix(pvc.Name, "clone-") {
```

Comment:
the condition could be reversed with return

Reply:
done